### ISSUE TITLE
Job Board: Add date information to layout

### ISSUE THREAD

**Issue by exoticDFT** [vsoch/usrse.github.io#30]
Currently we do not have information about the deadlines for application or when the job was posted visible in the layout. This would help those interested in the available jobs know which have upcoming deadlines and what is new/old.

**Comment by mtbc** [comment]
Note also https://github.com/USRSE/usrse.github.io/issues/348. Can imagine arguments for sorting by either.

**Comment by vsoch** [comment]
I think it's pretty rare that job listings have hard deadlines - at least from my small experience it seems like they are "open until filled" so (at best) our expiration dates are just guesses. But maybe if we show the date they were posted that gives a sense of how long they have been around.

**Comment by mtbc** [comment]
Posting date is especially useful for people who periodically come back and just read down from the top until they start looking familiar.

**Comment by exoticDFT** [comment]
You are right about most of the jobs not having expiration dates. I've just seen some in the past (my last job was like this.) May be worth trying a layout with more information and remove things if it's too "busy". Having a "since posted" label (e.g. "Posted 10 days ago") or just "posted on MM/DD/YY" label is the more crucial concern. Sorting based on these posted dates would be a great feature. And if an expiration date exists, we could reference it.

Either way, I definitely don't think we should create arbitrary deadlines, but we should figure out a way to handle dead posts. At least one on the board currently is no longer posted at the link we have. This may need to be another issue though.

**Comment by exoticDFT** [comment]
Uh actually I was just looking over the way we add job posts (had never looked before) and we specifically ask for an expiration date, but not a "posted date". So I guess we are handling jobs posts removing themselves after a while, but we are not handling dead posts. But it seems like we might want to add a new variable/label for posted date.

**Comment by exoticDFT** [comment]
It also seems like we may have a bug? We updated the jobs file last week to match us-rse.org, but https://vsoch.github.io/usrse.github.io/jobs/ and https://us-rse.org/jobs/ don't match up. And more specifically, the postings that aren't working seem to have expiration dates that have passed. One example, https://github.com/vsoch/usrse.github.io/blob/c72c67e75d0e98abfc77f29717936e0f270b4c64/_data/jobs.yml#L39-L42

**Comment by mtbc** [comment]
For https://github.com/vsoch/usrse.github.io/issues/30#issuecomment-844511365 the "posted date" I care about is when it was posted on our jobs site.

**Comment by exoticDFT** [comment]
> For [#30 (comment)](https://github.com/vsoch/usrse.github.io/issues/30#issuecomment-844511365) the "posted date" I care about is when it was posted on our jobs site.

@mtbc Just to clarify, by "our job site" you mean whatever job site the URL references? At the moment we are not asking for this information, so we'll have to start asking for a "posted" date along with the "expires" date. I just made some changes to the code to handled a "posted" date if it's included in the job entry.

**Comment by mtbc** [comment]
I just mean, on https://us-rse.org/jobs/ - if I'm looking at a reverse-chronological ordering of job postings on the US-RSE site and wanting to see which I've not yet seen, I want the posting date to be when it was added to the US-RSE site, so oughtn't need manual entry and lets me stop reading once they start looking familiar.

**Comment by exoticDFT** [comment]
Thanks for the clarification!

With the current system, I'm not sure how to guarantee a choronlogical order in the current way of handling new posts. At least in terms of "knowing" when a post was added. Currently, we have everyone edit a single file that contains all the posts. They are then rendered on the site based on the order in the file, with the top element being the first on our jobs page. In general, it seems people add new content to the top, but no guarantees, unless we explicitly state this in the guide and verify it during a PR.

My thought is to add the extra bit of information when posting a new job and we can sort based on that. Other suggestions are definitely welcome.

**Comment by mtbc** [comment]
If the posting date's missing, perhaps the job that removes the expired posts could also fill in its run date as the posting date? It'd be nice if the chronological sort were stable for equal dates.

**Comment by vsoch** [comment]
We can just do git blame or similar to get the dates they were added. Also - keep in mind that right now we need to focus on the big stuff. The sorting and even how we render jobs is probably something to do after that.

**Comment by vsoch** [comment]
@exoticDFT do you need any help here?

**Comment by exoticDFT** [comment]
> 
> 
> @exoticDFT do you need any help here?

Sorry for the delayed response.

I believe what I have in the branch update/job-dates has the framework to at least render/display posting dates (as shown in one of the above comments). Although, I'll obviously need to rebase to get the updates to master.

I think the last conversation we had about this was to wait on the "big stuff". So I guess part of the question is have we done the big stuff yet and should we have a conversation on how we want to add these posting dates? Do we want to create some automation for adding a "posted" variable to each item in the jobs.yml, do we want to ask users to start including this variable, or is there another method we should use?

**Comment by vsoch** [comment]
@exoticDFT this issue isn't labeled post-live, so we want to get this done before release I think. If you are able, please open the PR!

**Comment by vsoch** [comment]
And let's chat about your question in the meeting today!
